pages/api/module/post.ts

```diff
@@ -0,0 +1,43 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { ErrorResponse } from "types/ErrorResponse";
+import { Module, ModuleInterface } from "../../../database/models/module";
+import { Tool } from "../../../database/models/tool";
+import connectDB from "../utils/mongoose";
+
+const post = async (
+    req: NextApiRequest,
+    res: NextApiResponse<ModuleInterface | ErrorResponse>,
+): Promise<void> => {
+    if (!req.body.title)
+        return res
+            .status(404)
+            .send({ error: "Please provide your module with a title." });
+    const module = new Module({
+        title: req.body.title,
+        toolID: req.body.toolID,
+        slideIDs: req.body.slideIDs,
+        status: req.body.status,
+        editing: req.body.editing,
+    });
+    if (module.toolID) {
+        const tool = await Tool.findById(module.toolID).exec();
+        if (tool) {
+            if (!tool.moduleID) {
```

Comment:
isn't it fine to override the module id for a tool? Maybe we should ask @mkanoria

Reply:
Yes, that makes sense. Has been fixed @nimwijetunga .